src/Packetery/Module/Order/LabelPrint.php

```diff
@@ -309,33 +309,37 @@ private function requestPacketaLabels( int $offset, array $packetIds ): ?Respons
 		$request  = new Request\PacketsLabelsPdf( array_values( $packetIds ), $this->getLabelFormat(), $offset );
 		$response = $this->soapApiClient->packetsLabelsPdf( $request );
 		// TODO: is possible to merge following part of requestPacketaLabels and requestCarrierLabels?
-		$record         = new Log\Record();
-		$record->action = Log\Record::ACTION_LABEL_PRINT;
-		if ( ! $response->hasFault() ) {
-			foreach ( array_keys( $packetIds ) as $orderId ) {
+
+		foreach ( array_keys( $packetIds ) as $orderId ) {
+			$record         = new Log\Record();
+			$record->action = Log\Record::ACTION_LABEL_PRINT;
+
+			if ( ! $response->hasFault() ) {
 				$order = $this->orderRepository->getById( $orderId );
-				if ( null === $order ) { // Collection was already limited, so we do not need to log this.
-					continue;
+				if ( null !== $order ) {
+					$order->setIsLabelPrinted( true );
+					$this->orderRepository->save( $order );
 				}
-				$order->setIsLabelPrinted( true );
-				$this->orderRepository->save( $order );
+
+				$record->status = Log\Record::STATUS_SUCCESS;
+				$record->title  = __( 'labelPrintSuccessLogTitle', 'packetery' );
 			}
 
-			$record->status = Log\Record::STATUS_SUCCESS;
-			$record->title  = __( 'labelPrintSuccessLogTitle', 'packetery' );
-		} else {
-			$record->status = Log\Record::STATUS_ERROR;
-			$record->title  = __( 'labelPrintErrorLogTitle', 'packetery' );
-			$record->params = [
-				'request'      => [
-					'packetIds' => implode( ',', $request->getPacketIds() ),
-					'format'    => $request->getFormat(),
-					'offset'    => $request->getOffset(),
-				],
-				'errorMessage' => $response->getFaultString(),
-			];
+			if ( $response->hasFault() ) {
```

Comment:
Nedělej to jako další podmínku if, ale jako else té předchozí. Tak jak to bylo před tím.